modules/videoio/src/cap_android_mediandk.cpp

```diff
@@ -0,0 +1,247 @@
+/*
+ *
+ */
+
+#include "precomp.hpp"
+
+#include <stdio.h>
+#include <string.h>
+#include <fstream>
+#include <iostream>
+#include <unistd.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <fcntl.h>
+#include <android/log.h>
+
+#include "media/NdkMediaCodec.h"
+#include "media/NdkMediaExtractor.h"
+
+#define INPUT_TIMEOUT_MS 2000
+
+#define COLOR_FormatYUV420Planar 19
+#define COLOR_FormatYUV420SemiPlanar 21
+
+using namespace cv;
+
+#define TAG "NativeCodec"
+#define LOGV(...) __android_log_print(ANDROID_LOG_VERBOSE, TAG, __VA_ARGS__)
+#define LOGE(...) __android_log_print(ANDROID_LOG_ERROR, TAG, __VA_ARGS__)
+
+class AndroidMediaNdkCapture : public IVideoCapture
+{
+
+public:
+    AndroidMediaNdkCapture():
+        mediaExtractor(NULL), mediaCodec(NULL), sawInputEOS(false), sawOutputEOS(false),
+        frameWidth(0), frameHeight(0), colorFormat(0), buffer(NULL), bufferSize(0) {}
+    AMediaExtractor* mediaExtractor;
+    AMediaCodec *mediaCodec;
+    bool sawInputEOS;
+    bool sawOutputEOS;
+    int32_t frameWidth;
+    int32_t frameHeight;
+    int32_t colorFormat;
+    uint8_t* buffer;
+    size_t bufferSize;
```

Comment:
Consider using of `std::vector<uint8_t> buffer;` to avoid manual memory handling (and memory leaks).